src/query/service/src/sql/optimizer/rule/rewrite/rule_push_down_sort_scan.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cmp;
+
+use common_exception::Result;
+
+use crate::sql::optimizer::rule::Rule;
+use crate::sql::optimizer::rule::TransformState;
+use crate::sql::optimizer::RuleID;
+use crate::sql::optimizer::SExpr;
+use crate::sql::plans::LogicalGet;
+use crate::sql::plans::PatternPlan;
+use crate::sql::plans::RelOp;
+use crate::sql::plans::RelOperator;
+use crate::sql::plans::Sort;
+
+/// Input:  Limit
+///           \
+///          LogicalGet
+///
+/// Output:
+///           LogicalGet
+
+pub struct RulePushDownSortScan {
+    id: RuleID,
+    pattern: SExpr,
+}
+
+impl RulePushDownSortScan {
+    pub fn new() -> Self {
+        Self {
+            id: RuleID::PushDownSortScan,
+            pattern: SExpr::create_unary(
+                PatternPlan {
+                    plan_type: RelOp::Sort,
+                }
+                .into(),
+                SExpr::create_leaf(
+                    PatternPlan {
+                        plan_type: RelOp::LogicalGet,
+                    }
+                    .into(),
+                ),
+            ),
+        }
+    }
+}
+
+impl Rule for RulePushDownSortScan {
+    fn id(&self) -> RuleID {
+        self.id
+    }
+
+    fn apply(&self, s_expr: &SExpr, state: &mut TransformState) -> Result<()> {
+        println!(
```

Comment:
.

Reply:
Already delete the println at local code. I will push it when fix all comment.